src/binary-reader-objdump.cc

```diff
@@ -824,11 +824,22 @@ Result BinaryReaderObjdump::OnRelocCount(uint32_t count,
 Result BinaryReaderObjdump::OnReloc(RelocType type,
                                     uint32_t offset,
                                     uint32_t index,
-                                    int32_t addend) {
+                                    uint32_t addend) {
   uint32_t total_offset =
       section_starts[static_cast<size_t>(reloc_section)] + offset;
-  PrintDetails("   - %-18s idx=%#-4x addend=%#-4x offset=%#x(file=%#x)\n",
-               get_reloc_type_name(type), index, addend, offset, total_offset);
+  PrintDetails("   - %-18s offset=%#08x(file=%#08x) index=%#08x",
+               get_reloc_type_name(type), offset, total_offset, index);
+  if (addend) {
+    int32_t signed_addend = static_cast<int32_t>(addend);
+    if (signed_addend < 0) {
+      PrintDetails("-");
+      signed_addend = -signed_addend;
```

Comment:
I think this is UB when signed_addend is -0x80000000. Probably not possible to generate this case, though.